File: new-algorithm/src/javaalgo/haulzzang/week1/boj2164.java
package javaalgo.haulzzang.week1;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;

public class boj2164 {

    static int N;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {
        input();
        solution();
        System.out.println(sb.toString());
    }

    static void solution() {
        Queue<Integer> queue = new LinkedList<>();

        for (int i = 1; i <= N; i++) {
            queue.offer(i);
        }

        while (queue.size() != 1) {
            queue.poll();

            queue.add(queue.poll());
        }

        sb.append(queue.poll());
    }

    static void input() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        N = Integer.parseInt(br.readLine());
    }
}

File: new-algorithm/src/kotlinalgo/queue/boj2164.kt
package kotlinalgo.queue

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.LinkedList
import java.util.Queue

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val N = Integer.parseInt(readLine())

    val queue: Queue<Int> = LinkedList()

    for (i in 1..N) {
        queue.add(i)
    }

    while (queue.size > 1) {
        queue.remove()
        queue.add(queue.poll())
    }

    println(queue.poll())

}